frontend/src/hooks/useAsyncWithThrow.tsx

```
@@ -0,0 +1,49 @@
+import { DependencyList } from "react";
+import { useAsync } from "react-use";
+import {
+  FunctionReturningPromise,
+  PromiseType,
+} from "react-use/lib/misc/types";
+
+import { toError } from "../utils/ResponseUtil";
+
+export declare type AsyncState<T> =
+  | {
+      loading: boolean;
+      value?: undefined;
+    }
+  | {
+      loading: true;
+      value?: T;
+    }
+  | {
+      loading: false;
+      value?: undefined;
+    }
+  | {
+      loading: false;
+      value: T;
+    };
+
+// A thin wrapper of useAsync() in react-use, but it will throw an error
+export const useAsyncWithThrow = <T extends FunctionReturningPromise>(
```

`useAsync()` in react-use returns an error by return value. I want throwing an error, then catch it on ErrorBoundary so I create a thin `useAsync()` wrapper throwing an error if occurred instead of including in return value.